lib/sandbox.js

```diff
@@ -757,6 +757,66 @@ Sandbox.prototype.revokeToken = function (token, cb) {
     return cb ? promise.nodeify(cb) : promise;
 };
 
+/**
+ * List versions of a given node module that are available on the platform
+ * 
+ * @param {Object} options - Options
+ * @param {String} options.name - Name of the node module
+ * @param {Function} [cb] - Optional callback function for node-style callbacks
+ * @returns {Promise} A Promise that will be fulfilled with the token
+ */
+Sandbox.prototype.listNodeModuleVersions = function (options, cb) {
+    var request = Superagent
+        .get(this.url + `/api/env/node/modules/${encodeURIComponent(options.name)}`)
+        .set('Authorization', 'Bearer ' + this.token);
+
+    var promise = Request(request)
+        .get('body');
+
+    return cb ? promise.nodeify(cb) : promise;
+};
+
+/**
+ * Ensure that a set of modules are available on the platform
+ * 
+ * @param {Object} options - Options
+ * @param {Array} options.modules - Array of { name, version } pairs
+ * @param {Function} [cb] - Optional callback function for node-style callbacks
+ * @returns {Promise} A Promise that will be fulfilled with an array of { name, version, state } objects
+*/
+Sandbox.prototype.ensureNodeModules = function (options, cb) {
+    var request = Superagent
+        .post(this.url + '/api/env/node/modules')
+        .send({ modules: options.modules })
+        .set('Authorization', 'Bearer ' + this.token);
+
+    var promise = Request(request)
+        .get('body');
+
+    return cb ? promise.nodeify(cb) : promise;
+};
+
+/**
+ * Trigger a rebuild of a set of node modules
+ * 
+ * @param {Object} options - Options
+ * @param {Array} options.modules - Array of { name, version } pairs
+ * @param {Function} [cb] - Optional callback function for node-style callbacks
+ * @returns {Promise} A Promise that will be fulfilled with an array of { name, version, state } objects
+*/
+Sandbox.prototype.rebuildNodeModules = function (options, cb) {
+    var request = Superagent
+        .post(this.url + '/api/env/node/modules')
+        .query({ reset: 1 })
```

Comment:
Should we just have an option to `ensureNodeModules` instead of a first class method? Presence of method will have people scratch their heads while the method is really meant for cluster admins only, not general population.